packages/twenty-front/src/modules/domain-manager/hooks/useLastAuthenticateWorkspaceDomain.ts

```diff
@@ -0,0 +1,36 @@
+import { useRecoilCallback, useRecoilValue } from 'recoil';
+import { lastAuthenticateWorkspaceDomainState } from '@/domain-manager/states/lastAuthenticateWorkspaceDomainState';
+import { domainConfigurationState } from '@/domain-manager/states/domainConfigurationState';
+import { isDefined } from '~/utils/isDefined';
+
+export const useLastAuthenticateWorkspaceDomain = () => {
+  const domainConfiguration = useRecoilValue(domainConfigurationState);
+
+  const setLastAuthenticateWorkspaceDomain = useRecoilCallback(
```

Comment:
from the caller side

If you want:

Case 1:
const setLastAuthenticatedWorkspaceDomain = useLastAuthenticateWorkspaceDomain()
...
setLastAuthenticatedWorkspaceDomain({ workspaceId, subdomain })
==> you'll need to use a recoilcallback

Reply:
const setLastAuthenticatedWorkspaceDomain = useSetRecoilState()